src/java.base/share/classes/com/sun/crypto/provider/DESKey.java

```diff
@@ -108,26 +109,24 @@ public String getFormat() {
      * Calculates a hash code value for the object.
      * Objects that are equal will also have the same hashcode.
      */
+    @Override
     public int hashCode() {
-        int retval = 0;
-        for (int i = 1; i < this.key.length; i++) {
```

Comment:
I don't really know why the hashCode impl skipped the first byte of the array. Source code history doesn't give any insight. I thought maybe this had something to do with the parity bits (8 bits of a DES key are used for parity) but those are every 8th bit, and not all in the first byte. Best I can guess is that this is just an oversight and it should have included all bytes in the hash code calculation.

Reply:
Thanks for checking.